spacy_llm/tasks/ner.py

```diff
@@ -59,34 +59,51 @@ def _unique(items: Iterable[str]) -> Iterable[str]:
 @registry.llm_tasks("spacy.NER.v1")
```

Comment:
Since these changes are modifying the config, this should be become `spacy.NER.v2` and the package should maintain a copy of `spacy.NER.v1` with its original definition.

Reply:
hmm on principal I agree but this is an optional argument and the changing of the default template is internal. We could move this argument to the end of the `__init__` signature as well in case somebody was instantiating this task manually. It seems pretty low stakes to me to keep this as `spacy.NER.v1`. 

If we do make this `v2` then I assume we'd need a legacy package right?

Reply:
The saved configs are always auto-filled, so it has a default but it's not exactly optional. The problem is that someone with a saved config that includes the new setting couldn't load their config with an older release of the library even when it looks like the exact same registered function is available.

So you always need to modify the registered function name if you're making any changes to the set of config settings. Whether you'd need to increment the version number for changes in behavior is a separate question. Here I might be surprised as a user if I upgraded the library and suddenly got wildly different results for my NER task because the prompt was changed so much in the background, so it might be better to increment the version even if the config settings were the same. For minor bug fixes or modifications in behavior we wouldn't always modify the name, though, but we'd only make this change on a minor version of the library. (This is also related to how `spacy package` fills in requirements by default, which is always one minor version range.)

You wouldn't necessarily need a legacy package. For now it could just be included here in parallel, similar to the registered functions in `spacy-transformers` or `spacy-loggers`.

Reply:
Ok great points! thanks! I've named this new task `v2` and have the original template in its own factory function. 
Before we actually release this v2 officially I'd like to at least have this PR merged as well: https://github.com/explosion/spacy-llm/pull/81 so we don't need to do a `v3` task.